File: lib/features/pokemon/domain/usecases/get_pokemon_list_usecase.dart
import 'package:pokeapi/core/domain/entities/pokemon_entity.dart';
import 'package:pokeapi/core/domain/errors/pokemon_error.dart';
import 'package:pokeapi/features/pokemon/domain/repositories/pokemon_repository.dart';

/// Caso de uso para buscar a lista de Pokemons
class GetPokemonListUseCase {
  final PokemonRepository _repository;

  GetPokemonListUseCase(this._repository);

  /// Executa o caso de uso
  Future<Result<List<PokemonEntity>>> call({
    required int offset,
    required int limit,
  }) async {
    return _repository.getPokemonList(
      offset: offset,
      limit: limit,
    );
  }
}

File: bff/src/domain/usecases/get-pokemon-list.usecase.ts
import { Pokemon } from '../entities/pokemon.entity';
import { PokemonRepository } from '../repositories/pokemon.repository';

export class GetPokemonListUseCase {
  constructor(private readonly pokemonRepository: PokemonRepository) {}

  async execute(limit: number, offset: number): Promise<Pokemon[]> {
    return this.pokemonRepository.getPokemonList(limit, offset);
  }
}
